src/Abstractions/CharBoard.cs

```diff
@@ -72,4 +76,142 @@ public CharBoard Set(int r, int c, char value)
             return new CharBoard(Size, newCells);
         }
     }
+    
+    public record DiceBoard
+    {
+        private static readonly ConcurrentDictionary<int, DiceBoard> EmptyCache = new();
+        public static DiceBoard Empty(int size) => EmptyCache.GetOrAdd(size, size1 => new DiceBoard(size1));
+
+        public int Size { get; }
+        public ImmutableDictionary<int, Cell> Cells { get; }
```

Comment:
You don't need a dict if you use 100% index range from 0 to N :) You need an array :)